alpha/lib/model/entry.php

```diff
@@ -336,16 +343,145 @@ public function incViews ( $should_save = true )
 		myStatisticsMgr::incEntryViews( $this );
 	}
 
+	protected function getPartnerDefaultLanguage()
+	{
+		$partnerId = kCurrentContext::$partner_id ? kCurrentContext::$partner_id : kCurrentContext::$ks_partner_id;
+		$partner = PartnerPeer::retrieveByPK($partnerId);
+		return $partner->getDefaultLanguage();
+	}
+	
+	protected function handleMultiLanguageInput($field, &$originalValue, $update_db = true)
+	{
+		$defaultLanguage = $this->getPartnerDefaultLanguage();
+		$multiLangMapping = json_decode($this->getMultiLanguageMapping(), true);
+		$isMapUpdated = false;
+		$defaultValue = null;
+		$requestLanguage = kCurrentContext::getLanguage();
+		
+		if($defaultLanguage)
+		{
+			if (is_array($originalValue))
+			{
+				$defaultValue = $this->removeFromMapping($originalValue, $defaultLanguage);
+				if ($field == self::NAME && !$defaultValue && !$multiLangMapping)
```

Comment:
need to define validation function that in the future we will be able to run for any object we will want multi lang support on